x-pack/winlogbeat/module/routing/ingest/routing.yml

```diff
@@ -16,6 +16,17 @@ processors:
   - pipeline:
       name: '{< IngestPipeline "powershell_operational" >}'
       if: ctx?.winlog?.channel == 'Microsoft-Windows-PowerShell/Operational'
+  - set:
+      field: host.os.type
+      value: windows
+      override: false
+      if: ctx.tags != null && ctx.tags.contains('forwarded')
+  - set:
+      field: host.os.family
+      value: windows
+      override: false
+      if: ctx.tags != null && ctx.tags.contains('forwarded')
```

Comment:
Does the following config https://www.elastic.co/guide/en/beats/winlogbeat/master/configuration-winlogbeat-options.html#_event_logs_forwarded set the `forwarded` tag? I'm looking at the code and I don't think so.  So should we do both?

```
winlogbeat.event_logs:
  - name: <Channel>
    forwarded: true
    tags: [forwarded]
```

Reply:
Yeah, I spent a fair bit of time trying to sort that out. As far as I can make out, forwarded must be set by the user in the config as you have in the snippet. This is how it always appears in the repo.

Reply:
Would it not make sense to just if `forwarded: true` to automatically add the `forwarded` tag? That way users don't have to remember to do both?

Reply:
Or perhaps instead of requiring the tag. Just check if the field isn't set, set to `windows` otherwise use the provided field

Reply:
The config value doesn't make it's way into the ingested document AFAICS.

Reply:
Correct. I'm thinking in the winlogbeat code to either auto add the tag if you're going to leave the `if:....` conditional on the ingest processors OR change the conditional to just always set the fields if they're not already set since I think it's a decent default to assume the OS is windows

Reply:
> change the conditional to just always set the fields if they're not already set since I think it's a decent default to assume the OS is windows

I like this one.

Reply:
for our purposes until this is merged and released, it looks like i'm going to use the below in our environment
```
      - set:
          field: host.os.type
          value: windows
          override: false
      - set:
          field: host.os.family
          value: windows
          override: false
```
The override logic works so if its already set, nothing changes and if not, it gets set to windows.